megazords/ios-rust/build-xcframework.sh

```diff
@@ -118,30 +134,40 @@ rm -rf "$XCFRAMEWORK_ROOT"
 COMMON="$XCFRAMEWORK_ROOT/common/$FRAMEWORK_NAME.framework"
 
 mkdir -p "$COMMON/Modules"
-cp "$THIS_DIR/module.modulemap" "$COMMON/Modules/"
+cp "$WORKING_DIR/module.modulemap" "$COMMON/Modules/"
 
-cp "$THIS_DIR/DEPENDENCIES.md" "$COMMON/DEPENDENCIES.md"
+cp "$WORKING_DIR/DEPENDENCIES.md" "$COMMON/DEPENDENCIES.md"
 
 mkdir -p "$COMMON/Headers"
-cp "$THIS_DIR/MozillaRustComponents.h" "$COMMON/Headers"
+
+# First we move the files that are common between both
+# firefox-ios and Focus
+cp "$WORKING_DIR/$FRAMEWORK_NAME.h" "$COMMON/Headers"
 cp "$REPO_ROOT/components/rc_log/ios/RustLogFFI.h" "$COMMON/Headers"
 cp "$REPO_ROOT/components/viaduct/ios/RustViaductFFI.h" "$COMMON/Headers"
-cp "$REPO_ROOT/components/external/glean/glean-core/ffi/glean.h" "$COMMON/Headers"
-# TODO: https://github.com/mozilla/uniffi-rs/issues/1060
-# it would be neat if there was a single UniFFI command that would spit out
-# all of the generated headers for all UniFFIed dependencies of a given crate.
-# For now we generate the Swift bindings to get the headers as a side effect,
-# then delete the generated Swift code. Bleh.
 $CARGO uniffi-bindgen generate "$REPO_ROOT/components/nimbus/src/nimbus.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/crashtest/src/crashtest.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/fxa-client/src/fxa_client.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/logins/src/logins.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/autofill/src/autofill.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/push/src/push.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/tabs/src/tabs.udl" -l swift -o "$COMMON/Headers"
-$CARGO uniffi-bindgen generate "$REPO_ROOT/components/places/src/places.udl" -l swift -o "$COMMON/Headers"
+
+# We now only move/generate the rest of the headers if we are generating a full
+# iOS megazord
+if [ -z $IS_FOCUS ]; then
+  cp "$REPO_ROOT/components/external/glean/glean-core/ffi/glean.h" "$COMMON/Headers"
```

Comment:
Could've sworn Focus uses glean, but I think the swift consumers now get glean from it's own swift packages....but I think nimbus still uses the internal glean? We should probably follow up on this.

Reply:
Yeah they use glean through its own package and the internal usages will also end up using that package - the other concern would be local development, but the focus megazord shouldn't touch that, since that still uses the full xcframework

I was a little concerned something would be off, so I ran focus locally with this xcframework and verified that glean pings through nimbus get processed properly